### PR TITLE
Add derived .protean/project.json manifest with reconcile and drift detection

### LINKED ISSUE #1328: .protean/project.json manifest with reconcile and drift detection
### Gap

`.protean/` holds `config.toml` (IR compat config, via `src/protean/ir/config.py`) and `ir.json` (the materialized IR baseline, checked by `src/protean/ir/staleness.py`), but nothing records the project's own shape — its package name, layout, and the choices `protean new` made. There is no way to detect that the project on disk has drifted from what Protean expects.

### Decisions

- **Add `.protean/project.json`, a sibling to the existing `.protean/config.toml` and `.protean/ir.json`.** It records the project identity, layout, and scaffold version.
- **Derived and verifiable, never authoritative.** It records what the code says; a reconcile recomputes it and a drift check reports where disk and manifest disagree. It never overrides the code — editing the manifest to disagree with the code is reported as drift, not honored.
- **Reuse the `.protean/` loader pattern** from `ir/config.py` and the drift-report pattern from `ir/staleness.py`.

### Decision

- **Fuller manifest, but derived.** The manifest records adapter/provider choices in addition to identity + layout, establishing the precedent now so later work does not miss it. It stays **derived and reconcilable from `domain.toml`**, never hand-authored: the extra fields are populated from config and drift-checked, so the manifest never becomes a second source of truth.

### Acceptance criteria

- [ ] A generated project has `.protean/project.json`, loadable via the `.protean/` loader.
- [ ] A reconcile/drift check reports when the on-disk layout diverges from the manifest and makes no change on its own.
- [ ] Editing the manifest to disagree with the code is reported as drift, not honored (documented and tested).

### Deliverables

- New: the project-manifest schema, a writer invoked at `protean new` time, and a drift/reconcile check.

### Key files

- `src/protean/ir/config.py` (loader pattern), `src/protean/ir/staleness.py` (drift-report pattern), `src/protean/cli/new.py` (writes the manifest at generation).
- New: the project-manifest module.

### Tests

- Generate → manifest present and loads; introduce layout drift → reported; manifest-disagrees-with-code → reported as drift, not honored.

### Settled 2026-08-08

- **Moved to epic 10.6 (deterministic `add`, #1339).** Tracking and reconciling the project's shape as it evolves is an evolution concern; the manifest belongs with the `add`/change machinery that mutates the project, and is designed with it.
- **The manifest records only what it is the sole derived record for:** `package_name` (from the `src/<package>/` directory), `domain_name` (from `Domain(name=...)` in `domain.py`), and `layout` (the invariant subset fixed by ADR-0030 — composition root at `src/<package>/domain.py`, `domain.toml` beside it, element modules at most one directory below the root, `tests/` a sibling of `src/`). Each is recomputed from its own source; a manifest that disagrees with disk is drift. This is the correct long-term shape: one clean invariant ("derived, verifiable, never authoritative") that every field honours.
- **Nothing another file already owns is mirrored.** Adapter/provider choices stay in `domain.toml` (their source of truth); the manifest does not copy them, because a mirrored field protects no invariant — reconcile always matches `domain.toml`, so it can never meaningfully drift. `project_name` and `scaffold_version` move to `.copier-answers.yml`, which copier already writes as the generation-provenance record; they do not belong in a file whose contract is "derived and verifiable," because neither can be re-derived from disk after generation. Keeping the manifest purely derived is what makes the drift check honest.
- **Reuse `staleness.py`'s JSON loader** (`load_stored_ir` plus the mutation-free `StalenessResult` / `check_staleness` report), not `config.py`'s TOML loader. Note `.protean/` is not written by `protean new` today (config and baseline are materialized later), so the manifest writer is the first to create it.
- **This ships with an ADR** for the persisted `.protean/project.json` contract and its reconcile/drift semantics — or an in-scope extension of ADR-0030 — with the implementation.

### PR DESCRIPTION
**Pull request by subhashb**
Closes #1328.

Adds `.protean/project.json`, a derived and verifiable record of the project's own shape (identity and layout), with a reconcile that recomputes it from disk and a drift check that reports where disk and manifest disagree. The manifest is always derived from the code: a hand-edited manifest that disagrees with disk is reported as drift and never honored.

## What ships

- `src/protean/scaffold/manifest.py`: the feature. `ProjectManifest` / `ProjectLayout` frozen dataclasses with JSON round-trip; `reconcile_manifest` recomputes every field from disk (`package_name` from the single `src/*/domain.py`, `domain_name` from the AST-parsed `Domain(name=...)` literal, `layout` from the ADR-0030 invariants); `write_manifest` creates `.protean/` and writes sorted-key JSON; `check_manifest_drift` diffs disk against the stored manifest and returns MATCH / DRIFTED / NO_MANIFEST.
- `src/protean/cli/new.py`: one additive, `--pretend`-guarded `write_manifest` call at generation. No change to any signature, default, or behavior.
- `docs/adr/0033-project-manifest.md`: records the "derived, verifiable, never authoritative" contract and its reconcile/drift semantics, with the mkdocs nav entry.
- `changes/1328.added.md`: changelog fragment.
- `tests/scaffold/test_manifest.py`: 27 tests covering the three acceptance criteria, the negative and edge cases, and the CLI wiring, at 100% patch coverage on the new module.

## Acceptance criteria

- [x] A generated project has `.protean/project.json`, loadable via the `.protean/` loader.
- [x] A reconcile/drift check reports when the on-disk layout diverges from the manifest and makes no change on its own.
- [x] A hand-edited manifest that disagrees with the code is reported as drift and never honored (documented and tested).

### CODE REVIEW

File: src/protean/scaffold/manifest.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+            manifest_file=None,
+        )
+
+    stored_manifest, manifest_path = stored
+    recomputed = reconcile_manifest(root)
+
+    fields: list[tuple[str, str | None, str | None]] = [
+        ("package_name", stored_manifest.package_name, recomputed.package_name),
+        ("domain_name", stored_manifest.domain_name, recomputed.domain_name),
+        (
```
[the remark this answers is not in the document]
Comment (subhashb) [reply src/protean/scaffold/manifest.py:395]: Fixed. `check_manifest_drift` now compares `manifest_version` alongside the other fields, so a manifest written under an older schema (or hand-edited to a different version) reports as `DRIFTED` instead of `MATCH`. The function docstring says so too. Covered by a new test, `test_stale_manifest_version_alone_reads_as_drift`, where every other field still matches disk and the only divergence is `manifest_version` (stored `"0.9"` vs recomputed `MANIFEST_VERSION`).

File: tests/scaffold/test_manifest.py [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+    ProjectManifest,
+    _derive_domain_name,
+    check_manifest_drift,
+    load_stored_manifest,
+    reconcile_manifest,
+    write_manifest,
+)
+from tests.shared import isolated_filesystem
+
+runner = CliRunner()
```
[the remark this answers is not in the document]
Comment (subhashb) [reply tests/scaffold/test_manifest.py:31]: Fixed. Added `pytestmark = pytest.mark.no_test_domain` at the top of `tests/scaffold/test_manifest.py`, matching the sibling files `test_preview.py` and `test_change_plan.py`. These tests only build a tree in `tmp_path` and run the CLI, so they never needed the autouse `test_domain` fixture. All 28 tests still pass.

File: tests/scaffold/test_manifest.py [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+
+def test_hand_edited_manifest_reads_as_drift_and_disk_wins(tmp_path: Path) -> None:
+    _make_project(tmp_path, package="myproj", domain="myproj")
+    manifest_path = write_manifest(tmp_path)
+
+    # Hand-edit the stored manifest so every field disagrees with disk.
+    tampered = json.loads(manifest_path.read_text(encoding="utf-8"))
+    tampered["package_name"] = "hacked"
+    tampered["domain_name"] = "Hacked"
+    tampered["layout"]["composition_root"] = "src/hacked/domain.py"
```
[the remark this answers is not in the document]
Comment (subhashb) [reply tests/scaffold/test_manifest.py:499]: Fixed. `test_hand_edited_manifest_reads_as_drift_and_disk_wins` now also tampers `manifest_version` (to `"0.9"`) and asserts it is reported with the recomputed value equal to `MANIFEST_VERSION`. Added `test_stale_manifest_version_alone_reads_as_drift` too, where the version is the only stale field, so a regression that drops the version comparison fails even when nothing else diverges.

I left `test_non_literal_domain_name_reads_as_drift_against_stored_literal` (the other test you flagged) alone: it is about `domain_name` becoming underivable on disk, and the version-only case is covered by the dedicated test.
